addons/mail/static/src/js/services/mail_status_manager.js

```diff
@@ -54,13 +54,42 @@ MailManager.include({
     updateImStatus: function (statusList) {
         var updatedIDs = [];
         var self = this;
+        var dmChat;
+        var toUpdateOutOfOfficeChatIDs = [];
         _.each(statusList, function (status) {
             if (self._imStatus[status.id] === status.im_status) {
                 return;
             }
+            if (
+                status.im_status.indexOf('leave') !== -1 ||
+                (
+                    self._imStatus[status.id] &&
+                    self._imStatus[status.id].indexOf('leave') !== -1
+                )
+            ) {
+                dmChat = self.getDMChatFromPartnerID(status.id);
+                if (dmChat) {
+                    toUpdateOutOfOfficeChatIDs.push(dmChat.getID());
+                }
+            }
             updatedIDs.push(status.id);
             self._imStatus[status.id] = status.im_status;
         });
+        if (!_.isEmpty(toUpdateOutOfOfficeChatIDs)) {
+            this._rpc({
+                model: 'mail.channel',
+                method: 'channel_info',
+                args: [toUpdateOutOfOfficeChatIDs],
+            }).then(function (channelsInfo) {
+                _.each(channelsInfo, function (channelInfo) {
+                    dmChat = self.getChannel(channelInfo.id);
+                    dmChat.updateOutOfOfficeInfo({
```

Comment:
is there always a dmChat?

Reply:
yes, because this is limited to dmChats from `toUpdateOutOfOfficeChatIDs`, and this list contains IDs of existing dmChats